types/wlr_seat.c

```diff
@@ -874,11 +874,9 @@ void wlr_seat_keyboard_enter(struct wlr_seat *seat,
 
 		struct wl_array keys;
 		wl_array_init(&keys);
-		for (size_t i = 0; i < WLR_KEYBOARD_KEYS_CAP; ++i) {
-			if (keyboard->keycodes[i] != 0) {
-				uint32_t *p = wl_array_add(&keys, sizeof(uint32_t));
-				*p = keyboard->keycodes[i];
-			}
+		for (size_t i = 0; i < keyboard->num_keycodes; ++i) {
+			uint32_t *p = wl_array_add(&keys, sizeof(uint32_t));
```

Comment:
wl_array_add can return null

Reply:
Can you grep for wl_array_add and make sure that all return values are checked?

Reply:
Can you grep for wl_array_add and make sure that all return values are checked?

Reply:
see top comment, about half are. I'll submit a PR to fix all that don't I guess.